sql/parser/sql.y

```diff
@@ -1529,7 +1529,7 @@ transaction_stmt:
   }
 | BEGIN opt_transaction transaction_mode_list_or_empty
   {
-    $$ = nil
+    $$ = &BeginTransaction{}
```

Comment:
You should add some simple cases to `ParseTest` for `{BEGIN,COMMIT,ROLLBACK} TRANSACTION`.